src/include/duckdb/common/vector_operations/generic_executor.hpp

```diff
@@ -93,28 +116,40 @@ template <class A_TYPE, class B_TYPE>
 struct StructTypeBinary {
 	A_TYPE a_val;
 	B_TYPE b_val;
+	bool a_is_null;
+	bool b_is_null;
 
 	using STRUCT_STATE = StructTypeState<2>;
 
-	static bool ConstructType(STRUCT_STATE &state, idx_t i, StructTypeBinary<A_TYPE, B_TYPE> &result) {
+	bool ContainsNull() {
+		return a_is_null || b_is_null;
+	}
+
+	static StructTypeBinary<A_TYPE, B_TYPE> ConstructType(STRUCT_STATE &state, idx_t i) {
 		auto &a_data = state.child_data[0];
 		auto &b_data = state.child_data[1];
 
 		auto a_idx = a_data.sel->get_index(i);
 		auto b_idx = b_data.sel->get_index(i);
-		if (!a_data.validity.RowIsValid(a_idx) || !b_data.validity.RowIsValid(b_idx)) {
-			return false;
+		StructTypeBinary<A_TYPE, B_TYPE> result;
+		result.a_is_null = !a_data.validity.RowIsValid(a_idx);
```

Comment:
Same here - if we would use `ConstructType` recursively on the child types we wouldn't need to do this step here ourselves.